tests/i18n/test_extraction.py

```diff
@@ -777,6 +777,27 @@ def _test_project_locale_paths(self, locale_path):
                 self.assertMsgId('This app has a locale directory', po_contents)
 
 
+class LocaleValidityTests(ExtractorTests):
+
+    def test_locales_format_valid(self):
+        LOCALE = ['en_IN', 'th_TH_TH', 'asa_TZ', 'az_Cyrl', 'yue_Hant_HK', 'haw_US', 'ar_001']
+        for locale in LOCALE:
+            with self.subTest(locale):
+                out = StringIO()
+                management.call_command('makemessages', locale=[locale], verbosity=1, stdout=out)
+                self.assertNotIn('invalid', out.getvalue())
+                self.assertIn('processing locale %s' % locale, out.getvalue())
+
+    def test_locales_format_invalid(self):
+        LOCALE = ['en-IN', 'th-TH-in', 'aSA-tZ', 'aZ-CyRl', 'EN-IN', 'EN-in', 'en-USS']
```

Comment:
We don't need to test multiple cases because we want to ignore only locales with hyphens.

Reply:
Yeah! Earlier I felt like it would be great to have tests in separate classes but I think as we have to test just for a hyphen, there is no need for a separate class Thanks for the improvement.